include/circt/Dialect/MSFT/MSFTOps.td

```diff
@@ -240,3 +242,19 @@ def PhysLocationOp : MSFTOp<"pd.location"> {
     ($ref^)? custom<PhysLoc>($loc) (`path` `:` $subPath^)? attr-dict
   }];
 }
+
+def DynamicInstanceOp : MSFTOp<"instance.dynamic",
+                               [HasParent<"mlir::ModuleOp">, NoTerminator]> {
+
+  let summary = "A module instance in the instance hierarchy";
+  let arguments = (ins NameRefArrayAttr:$appid);
```

Comment:
For now, piggybacking on `NameRefArray` for appid.

Reply:
Makes sense. If we can, using the HW attributes should help make it easier to integrate some day.